x-pack/plugins/security_solution/public/common/hooks/use_data_view_id.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useSelector } from 'react-redux';
+import { sourcererScopeSelectedDataViewId } from '../store/sourcerer/selectors';
+import type { SourcererScopeName } from '../store/sourcerer/model';
+import type { State } from '../store';
+
+export const useDataViewId = (scopeId: SourcererScopeName): string | undefined => {
```

Comment:
is there really a value in making this a hook? If I'm not mistaken it's only used in a single place at the moment. The hook is very small so doesn't really save that much lines of code in the place where it's used. I feel like if we want to keep it extracted we should unit test it? (but it feels overkill to do so)

Reply:
Hey @PhilippeOberti, yes this hook is used in two places:

https://github.com/elastic/kibana/blob/e381c0da36f84e9527f171f182c7748fd4033070/x-pack/plugins/security_solution/public/common/components/hover_actions/use_hover_action_items.tsx#L89

and 

https://github.com/elastic/kibana/blob/e381c0da36f84e9527f171f182c7748fd4033070/x-pack/plugins/security_solution/public/detections/hooks/trigger_actions_alert_table/use_cell_actions.tsx#L28

I haven't added a unit test because if we mock the selector imported this hook has no logic to test, it is just a helper encapsulating the state selection, we would only be testing `useSelector` works. But I agree that this looks a bit untested, I'll check if I can add a test case in other related jest tests to verify this hook returns the expected value.

btw, we also have UI integration tests for the filter actions that execute this hook ([here](https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_cypress/cypress/e2e/investigations/alerts/alerts_cell_actions.cy.ts) and [here](https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_cypress/cypress/e2e/investigations/timelines/filters.cy.ts)).